src/components/Common/ButtonWithDropdown/index.tsx

```diff
@@ -39,21 +38,37 @@ const DropdownItem: React.FC<DropdownItemProps> = ({
   );
 };
 
-interface ButtonWithDropdownProps
-  extends ButtonHTMLAttributes<HTMLButtonElement> {
+// Helper type to override types (overrides onClick)
+type MergeElementProps<
+  T extends React.ElementType,
+  P extends Record<string, unknown>
+> = Omit<React.ComponentProps<T>, keyof P> & P;
+
+type ElementTypes = 'button' | 'a';
+
+type ButtonWithDropdownBaseProps<P> = {
   text: ReactNode;
   dropdownIcon?: ReactNode;
   buttonType?: 'primary' | 'ghost';
-}
+  // Had to do declare this manually as typescript would assume e was of type any otherwise
+  onClick?: (
+    e: React.MouseEvent<P extends 'a' ? HTMLAnchorElement : HTMLButtonElement>
+  ) => void;
+};
+
+type ButtonWithDropdownProps<P extends React.ElementType> = {
+  as?: P;
+} & MergeElementProps<P, ButtonWithDropdownBaseProps<P>>;
```

Comment:
We could just use any types that are identical from `Button` and import them.

Reply:
But they aren't exactly the same 🤔